web/streetmanager/srwr_process.php

```diff
@@ -0,0 +1,345 @@
+while ($data = fgetcsv($csv_file)) {
+      collateRows($data);
+}
+
+// Do any data changes and checks
+manipulateData();
+
+// Remove data marked to be dropped
+$printrow = _filterData();
+
+// Insert into the streetmanager database
+insertRows();
+
+function _filterData() {
+	global $printrow;
+
+	$filtered = array_filter($printrow, function($value, $key){
+		return !array_key_exists('drop_data', $value);
+	}, ARRAY_FILTER_USE_BOTH);
+
+	return $filtered;
+}
+
+function deleteFiles() {
+	global $path;
+
+	if (!file_exists($path)) {
+		mkdir($path, 0777, true);
+		return;
+	}
+
+	$files = glob($path . '*');
+	foreach($files as $file) {
+  		if(is_file($file)) {
+    		unlink($file);
+  		}
+	}
+}
+
+function fetchCSVFile() {
+	global $path;
+	$url = 'https://downloads.srwr.scot/export/api/v1/daily';
+
+	$ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    $response = curl_exec($ch);
+    $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    if ($httpcode != 200) {
+		die("Error: Fetching url for zip file failed");
+    } else {
+		$data = json_decode($response, true);
+		$url = $data['url'];
+		curl_setopt($ch, CURLOPT_URL, $url);
+		$response = curl_exec($ch);
+		if ($httpcode != 200) {
+			die("Error: Fetching zip file failed");
+    	} else {
+			file_put_contents($path . 'daily.zip', $response);
+		}
+    }
+    curl_close($ch);
+    $ch = null;
+}
+
+function unzipCSVFile() {
+	global $path;
+
+	$zip = new ZipArchive();
+	$zipfiles = glob($path . '*.zip');
+	if (sizeof($zipfiles) != 1) {
+		die("Error: Expecting one zip file");
+	} else {
+		$zip->open($zipfiles[0]);
+		$zip->extractTo($path);
+		$csvfiles = glob($path . '*.csv');
+		if (sizeof($csvfiles) != 1) {
+			die("Error: Expecting one csv file");
+		} else {
+			$csv_file = fopen($csvfiles[0], 'r');
+			return $csv_file;
+		}
+	}
+}
+
+function collateRows($row) {
+	 global $printrow;
+	 global $companies;
+
+	 $activityId = $row[2];
+	 if ($row[1] == '001') {
+		$printrow[$activityId]['latest_phase'] = $row[11];
+	 } else if ($row[1] == '006') {
+		$printrow[$activityId]['project_ref'] = $row[3];
+	 } else if ($row[1] == '007') {
+	    $printrow[$activityId]['works_location_coordinates'] = $row[12];
+	    $printrow[$activityId]['location_text'] = $row[6];
+		$printrow[$activityId]['work_category'] = _workCategory($row[9]);
+		$printrow[$activityId]['work_status'] = _activityStatuses($row[10]);
+	 } else if ($row[1] == '010') {
+	    $printrow[$activityId]['works_location_type'] = _siteLocations($row[9]);
+	 } else if ($row[1] == '008') {
+	   if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[6] ) {
+	      $printrow[$activityId]['proposed_start_date'] = _stripTime($row[6]);
+	   } else if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[13]) {
+	      $printrow[$activityId]['proposed_start_date'] = _stripTime($row[13]);
```

Comment:
Is there a reason to use "latest start date proposed" instead of "proposed start date time"?
(and same for end)